server_admin/topics/realms/keys.adoc

```diff
@@ -22,35 +22,79 @@ that is able to provide an active keypair.
+application, the SSO cookie is updated with the new signature. When OpenID Connect tokens are refreshed new tokens are
+signed with the new keys. Eventually, all cookies and tokens use the new keys and after a while the old keys can be removed.
 
 The frequency of deleting old keys is a tradeoff between security and making sure all cookies and tokens are updated. Consider creating new keys every three to six months and deleting old keys one to two months after you create the new keys. If a user was inactive in the period between the new keys being added and the old keys being removed, that user will have to re-authenticate.
 
 Rotating keys also applies to offline tokens. To make sure they are updated, the applications need to refresh the tokens before the old keys are removed.
 
 ==== Adding a generated keypair
 
+Use this procedure to generate a keypair including a self-signed certificate.
+
 .Procedure
 . Select the realm in the Admin Console.
 . Click *Realm settings* in the menu.
 . Click the *Keys* tab.
 . Click the *Providers* tab.
 . Click *Add provider* and select *rsa-generated*.
-. Enter a number in the *Priority* field. This number determines if the new key pair becomes the active key pair.
+. Enter a number in the *Priority* field. This number determines if the new keypair becomes the active keypair. The highest number makes the keypair active.
 . Select a value for *AES Key size*.
 . Click *Save*.
 
-This action will generated a new keypair including a self-signed certificate.
-
 Changing the priority for a provider will not cause the keys to be re-generated, but if you want to change the keysize
 you can edit the provider and new keys will be generated.
 
+==== Rotating keys by extracting a certificate
+
+You can rotate keys by extracting a certificate from an RSA generated keypair and using that certificate in a new keystore.
+
+.Prerequisites
+* A generated keypair 
+
+.Procedure
+. Select the realm in the Admin Console.
+. Click *Realm Settings*.
+. Click the *Keys* tab.
++
+A list of *Active* keys appears.
+. On a row with an RSA key, click *Certificate* under *Public Keys*.
++
+The certificate appears in text form.
+. Save the certificate to a file and enclose it in these lines.
++
+[Source,Bash,Subs=+Attributes]
+----
+----Begin Certificate---- 
+<Output>
+----End Certificate---- 
+----
+. Use the *keytool* command to convert the key file to PEM Format.
+
+. Remove the current RSA public key certificate from the keystore.
++
+[Source,Bash,Subs=+Attributes]
+----
+keytool -delete -keystore <keystore>.jks -storepass <password> -alias <key>
+----
+. Import the new certificate into the keystore
++
+[Source,Bash,Subs=+Attributes]
+----
+keytool -importcert -file domain.crt -keystore <keystore>.jks -storepass <password>  -alias <key>
+----
+. Undeploy and rebuild the application.
++
+[Source,Bash,Subs=+Attributes]
+----
+wildfly:undeploy
```

Comment:
WildFly was deprecated and no longer supported, as far as I'm aware.

Reply:
Thanks, @abstractj 
orivat@redhat.com Do you have an alternative for the step to undeploy and rebuild the application?